index.html

```diff
@@ -481,45 +485,102 @@
 
   var app = angular.module('app', []);
   app.controller('Ctrl', function ($scope, $location) {
+    $scope.numPlayers = 1;
     $scope.showSettings = false;
-    $scope.playerColor = "#FFFFFF";
-    $scope.showRegionPicker = false;
-
-    $scope.$watchGroup(["map", "home", "orig", "dest", "playerColor"], function(nv) {
-      // Reasonable defaults
-      var map   = nv[0] = (nv[0] === undefined) ? "US" : nv[0];
-      var home  = nv[1] = (nv[1] === undefined) ? -1 : +nv[1];
-      var orig  = nv[2] = (nv[2] === undefined) ? -1 : +nv[2];
-      var dest  = nv[3] = (nv[3] === undefined) ? -1 : +nv[3];
-      var color = nv[4] = (nv[4] === undefined) ? "white" : nv[4];
-
-      // Update map-related variables.
-      $scope.data = maps[map]
+    $scope.players = []
+
+    /**
+     * Watch for map changes.
+     */
+    $scope.$watch("map", function(map) {
+      map = (map === undefined) ? "US" : map;
+      $scope.data = maps[map];
       $scope.cities = $scope.data.cities;
       $scope.regions = $scope.data.regions;
 
-      // Update regions to match user selection.
-      $scope.homeRegion = $scope.data.city_regions[home];
-      $scope.origRegion = $scope.data.city_regions[orig];
-      $scope.destRegion = $scope.data.city_regions[dest];
+      // Update all of the player data for the new map
+      $scope.updatePlayers();
 
-      // Update the payoff.
-      if ($scope.orig < 0 || $scope.dest < 0) {
-        $scope.payoff = "n/a";
-      } else {
-        $scope.payoff = $scope.data.payoff[orig][dest];
+      // Update the location history
+      $location.search("map", map);
+    });
+
+    /**
+     * Watch for player city changes.
+     */
+    $scope.$watch("players", function(players) {
+      $scope.updatePlayers();
```

Comment:
If you put this code after your definition of updatePlayers, it wouldn't be necessary to wrap it in a function. eg., you could have: $scope.$watch("players", $scope.updatePlayers, true);